zvariant/src/gvariant/de.rs

```diff
@@ -274,7 +274,7 @@ impl<'de, 'd, 'sig, 'f, #[cfg(unix)] F: AsFd, #[cfg(not(unix))] F> de::Deseriali
     where
         V: Visitor<'de>,
     {
-        visitor.visit_unit()
+        self.deserialize_unit(visitor)
```

Comment:
Separate logical change most definitely since it's nothing to do with the tests and commit title doesn't cover this change. Please split into its own commit with it's own title and details.